### ISSUE TITLE
Add configuration option to specify multiline_match_regexp  instead of _end_regexp or flush interval

### ISSUE THREAD

**Issue by sjoeboo** [fluent-plugins-nursery/fluent-plugin-concat#31]
I have a high volume log (fast), with single and multiline messages coming in. Multiline messages are always together, however (thankfully). Right now, I'm using concat like:

```
<filter sp.*>
  @type concat
  key msg
  multiline_start_regexp /Multiline_Start/
    flush_interval 1s
    timeout_label "@NORMAL"
</filter>
```

and it works. However, if a normal, non-multiline message comes in before the flush_interval, it is also appended, obviously not ideal. 
I would LOVE to use the _end_regexp, but my multiline message don't have an obvious end (all but the first line is indented with a tab, however). 

So I was hoping to get something like `multiline_match_regexp` `/^\t`.
When used with the start_regexp, the concat would consist of the first message (start_regexp), and all subseqent lines *that match the `multiline_match_regexp` only*. Once a line comes in that does not match the `multiline_match_regexp`, flush. 

This would help me avoid gathering the wrong lines together with a flush interval, and properly detect the termination of my multiline (whenever a line comes in that doesn't match the multiline_match_regexp) 

it doesn't seem like that much of a stretch...? 

also thanks for the great plugin!

**Comment by okkez** [comment]
Could you try #32 ? @sjoeboo 

If you are ok, I will merge #32 and release new version.
Otherwise:

* could you paste concrete log and configurations?
* could you describe expected result?

**Comment by sjoeboo** [comment]
@okkez wow thanks! yep just tested and it works and is exactly what i was looking for! thanks so much!

**Comment by okkez** [comment]
Released 2.1.0
